packages/backend/src/services/scanner.py

```
@@ -114,10 +141,25 @@ def scan_library(library_id: str, root: str) -> int:
 # --- Async scanning with progress tracking ---
 _scan_lock = Lock()
 _current_scans: set[str] = set()
+_cancel_scans: set[str] = set()
+_scan_threads: dict[str, "threading.Thread"] = {}
```

Ruff (0.14.10)

Undefined name `threading`

(F821)